File: sort revision/selection_sort.cpp
#include<iostream>
using namespace std;


void selection(int arr[],int n)
{
    int temp[n];

    for(int i=0;i<n;i++)
    {
        int min_idx = 0;
        for(int j=1;j<n;j++)
        {
            if(arr[j]<arr[min_idx])
            {
                min_idx = j;
            }
        }

        temp[i] = arr[min_idx];
        arr[min_idx] = INFINITY;
    }

    for(int i=0;i<n;i++)
    {
        arr[i] = temp[i]; 
    }

    for(int i=0;i<n;i++)
    {
        cout<<temp[i]<<" ";
    }
}


int main()
{
    int arr[] = {2,1,5,4,3};
    int n = 5;

    selection(arr,n);
    return 0;
}

File: sort revision/selection_sort.js
let selectionSort = (arr,n)=>{
    let temp = [];
    for(let i=0;i<n;i++){
        let min_idx = 0;
        for(let j = 1;j<n;j++){
            if(arr[j]<arr[min_idx]){
                min_idx = j;
            }
        }
        temp[i] = arr[min_idx];
        arr[min_idx] = Number.MAX_VALUE;
    }

    console.log(temp)

}

function main(){
    let arr = [3,2,5,4,1];
    selectionSort(arr,arr.length);
}

main();